src/main/kotlin/util/racing/generator/NumberGenerator.kt

```diff
@@ -0,0 +1,7 @@
+package util.racing.generator
+
+@FunctionalInterface
+interface NumberGenerator {
+
+    fun generate(): Int
+}
```

Comment:
FunctionalInterface 어노테이션 사용이 불필요해보입니다. 함수형 인터페이스를 만들고 싶으시면 fun 키워드를 붙이는 것으로도 충분합니다. 함수형 인터페이스를 활용하면 불필요한 object 키워드를 통한 인스턴스 생성 구문을 줄일 수 있습니다.
```suggestion
fun interface NumberGenerator {

    fun generate(): Int
}
```

[사용법 예시]
```kotlin
// 기존
val isEven = object : IntPredicate {
   override fun accept(i: Int): Boolean {
       return i % 2 == 0
   }
}
// 함수형 인터페이스
val isEven = IntPredicate { it % 2 == 0 }
```

참고: https://kotlinlang.org/docs/fun-interfaces.html

Reply:
오 더 간단하네요 :) 감사합니다.